exporters/otlp/internal/transform/attribute_test.go

```diff
@@ -146,3 +148,147 @@ func TestAttributes(t *testing.T) {
+	// "bool", "int", "int32", "int64",
+	// "float32", "float64", "string",
+	// "uint", "uint32", "uint64"
+	for _, test := range []attributeTest{
+		{nil, nil},
+		{
+			[]kv.KeyValue{
+				kv.Array("bool array to bool array", []bool{true, false}),
+				kv.Array("int array to int64 array", []int{1, 2, 3}),
+				kv.Array("uint array to int64 array", []uint{1, 2, 3}),
+				kv.Array("int32 array to int64 array", []int32{1, 2, 3}),
+				kv.Array("uint32 array to int64 array", []uint32{1, 2, 3}),
+				kv.Array("int64 array to int64 array", []int64{1, 2, 3}),
+				kv.Array("uint64 array to int64 array", []uint64{1, 2, 3}),
+				kv.Array("float32 array to double array", []float32{1.11, 2.22, 3.33}),
+				kv.Array("float64 array to double array", []float64{1.11, 2.22, 3.33}),
+				kv.Array("string array to string array", []string{"foo", "bar", "baz"}),
+			},
+			[]*commonpb.KeyValue{
+				newOtelBoolArray("bool array to bool array", []bool{true, false}),
+				newOtelIntArray("int array to int64 array", []int64{1, 2, 3}),
+				newOtelIntArray("uint array to int64 array", []int64{1, 2, 3}),
+				newOtelIntArray("int32 array to int64 array", []int64{1, 2, 3}),
+				newOtelIntArray("uint32 array to int64 array", []int64{1, 2, 3}),
+				newOtelIntArray("int64 array to int64 array", []int64{1, 2, 3}),
+				newOtelIntArray("uint64 array to int64 array", []int64{1, 2, 3}),
+				newOtelDoubleArray("float32 array to double array", []float64{1.11, 2.22, 3.33}),
+				newOtelDoubleArray("float64 array to double array", []float64{1.11, 2.22, 3.33}),
+				newOtelStringArray("string array to string array", []string{"foo", "bar", "baz"}),
+			},
+		},
+	} {
+		actualArrayAttributes := Attributes(test.attrs)
+		expectedArrayAttributes := test.expected
+		if !assert.Len(t, actualArrayAttributes, len(expectedArrayAttributes)) {
+			continue
+		}
+
+		for i, actualArrayAttr := range actualArrayAttributes {
+			expectedArrayAttr := expectedArrayAttributes[i]
+			if actualArrayAttr.Key != expectedArrayAttr.Key {
+				continue
+			}
+
+			expectedArrayValue := expectedArrayAttr.Value.GetArrayValue()
+			assert.NotNil(t, expectedArrayValue)
+
+			actualArrayValue := actualArrayAttr.Value.GetArrayValue()
+			assert.NotNil(t, actualArrayValue)
+
+			assertExpectedArrayValues(t, expectedArrayValue.Values, actualArrayValue.Values)
+		}
+
+	}
+}
+
+func assertExpectedArrayValues(t *testing.T, expectedValues, actualValues []*commonpb.AnyValue) {
+	for i, actual := range actualValues {
+		expected := expectedValues[i]
+		if a, ok := actual.Value.(*commonpb.AnyValue_DoubleValue); ok {
+			e, ok := expected.Value.(*commonpb.AnyValue_DoubleValue)
+			if !ok {
+				t.Errorf("expected AnyValue_DoubleValue, got %T", expected.Value)
+				continue
+			}
+			if !assert.InDelta(t, e.DoubleValue, a.DoubleValue, 0.01) {
+				continue
+			}
+			e.DoubleValue = a.DoubleValue
+		}
+		assert.Equal(t, expected, actual)
+	}
+}
+
+func newOtelBoolArray(key string, values []bool) *commonpb.KeyValue {
```

Comment:
nit, OTel is an initialism and the capitalization should be preserved.

```suggestion
func newOTelBoolArray(key string, values []bool) *commonpb.KeyValue {
```

Reply:
Changed